mithril-aggregator/src/database/repository/pending_certificate_repository.rs

```diff
@@ -0,0 +1,202 @@
+                fake_data::protocol_parameters(),
+                fake_data::signers(4),
+                fake_data::signers(5),
+            );
+
+            store.save(certificate_pending).await.unwrap();
+        }
+        store
+    }
+
+    #[tokio::test]
+    async fn get_certificate_pending_with_existing_certificate() {
+        let store = get_certificate_pending_store(true).await;
+        let result = store.get().await.unwrap();
+
+        assert!(result.is_some());
+    }
+
+    #[tokio::test]
+    async fn get_certificate_pending_with_no_existing_certificate() {
+        let store = get_certificate_pending_store(false).await;
+        let result = store.get().await.unwrap();
+
+        assert!(result.is_none());
+    }
+
+    #[tokio::test]
+    async fn save_certificate_pending_once() {
+        let store = get_certificate_pending_store(false).await;
+        let signed_entity_type = SignedEntityType::dummy();
+        let certificate_pending = CertificatePending::new(
+            Epoch(2),
+            signed_entity_type,
+            fake_data::protocol_parameters(),
+            fake_data::protocol_parameters(),
+            fake_data::signers(1),
+            fake_data::signers(2),
+        );
+
+        assert!(store.save(certificate_pending).await.is_ok());
+        assert!(store.get().await.unwrap().is_some());
+    }
+
+    #[tokio::test]
+    async fn update_certificate_pending() {
+        let store = get_certificate_pending_store(true).await;
+        let certificate_pending = store.get().await.unwrap().unwrap();
+
+        assert!(store.save(certificate_pending).await.is_ok());
+    }
+
+    #[tokio::test]
+    async fn remove_certificate_pending() {
+        let store = get_certificate_pending_store(true).await;
+        let epoch = Epoch(0);
+        let certificate_pending = store.remove().await.unwrap().unwrap();
+
+        assert_eq!(epoch, certificate_pending.epoch);
+        assert!(store.get().await.unwrap().is_none());
+    }
+
+    #[tokio::test]
+    async fn should_migrate_data_from_adapter() {
+        let certificate_pending = CertificatePending::new(
+            Epoch(0),
+            SignedEntityType::dummy(),
+            fake_data::protocol_parameters(),
+            fake_data::protocol_parameters(),
+            fake_data::signers(4),
+            fake_data::signers(5),
+        );
+
+        let migrations = crate::database::migration::get_migrations();
+
+        let connection = Arc::new(ConnectionBuilder::open_memory().build().unwrap());
+        let pending_certificate_adapter =
+            FakeStoreAdapter::new(connection.clone(), "pending_certificate");
+        pending_certificate_adapter.create_table();
+
+        ConnectionBuilder::open_memory()
+            .apply_migrations(
+                &connection,
+                migrations
+                    .clone()
+                    .into_iter()
+                    .filter(|m| m.version < 32)
```

Comment:
Just a reminder if you want to update the version number after rebasing on the main branch.

Reply:
Done